frontend/public/components/resource-quota.jsx

```diff
@@ -35,64 +56,118 @@ const Row = ({obj: rq}) => <div className="row co-resource-list__item">
+  } else if (percent >= 50 && percent < 100){
+    usageIconClass = 'pficon pficon-resources-almost-full';
+  } else if (percent === 100) {
+    usageIconClass = 'pficon pficon-resources-full';
+  } else if (percent > 100) {
+    usageIconClass = 'pficon pficon-warning-triangle-o';
   }
-  return <i className={proportionIconClass} aria-hidden="true" />;
-};
-
-const getProportionUsed = (used, max) => {
-  if (!max || !used) {
-    return 0;
-  }
-
-  const usedNum = convertToBaseValue(used);
-  const maxNum = convertToBaseValue(max);
-  return usedNum / maxNum;
+  return <i className={usageIconClass} aria-hidden="true" />;
 };
 
 const ResourceUsageRow = ({quota, resourceType}) => {
-  const max = _.get(quota, ['status', 'hard', resourceType]) || _.get(quota, ['spec', 'hard', resourceType]);
-  const used = _.get(quota, ['status', 'used', resourceType]);
-  const proportionUsed = getProportionUsed(used, max);
+  const { used, max, percent } = getResourceUsage(quota, resourceType);
   return <div className="row co-m-row">
     <div className="col-sm-4 col-xs-6 co-break-word">{resourceType}</div>
-    <div className="col-sm-2 hidden-xs co-resource-quota-icon"><ProportionIcon proportion={proportionUsed} /></div>
+    <div className="col-sm-2 hidden-xs co-resource-quota-icon"><UsageIcon percent={percent} /></div>
     <div className="col-sm-3 col-xs-3">{used}</div>
     <div className="col-sm-3 col-xs-3">{max}</div>
   </div>;
 };
 
-const Details = ({obj: rq}) => <React.Fragment>
-  <div className="co-m-pane__body">
-    <SectionHeading text="Resource Quota Overview" />
-    <ResourceSummary resource={rq} showPodSelector={false} showNodeSelector={false} />
-  </div>
-  <div className="co-m-pane__body">
-    <SectionHeading text="Resource Usage" />
-    <div className="co-m-table-grid co-m-table-grid--bordered">
-      <div className="row co-m-table-grid__head">
-        <div className="col-sm-4 col-xs-6">Resource Type</div>
-        <div className="col-sm-2 hidden-xs">Capacity</div>
-        <div className="col-sm-3 col-xs-3">Used</div>
-        <div className="col-sm-3 col-xs-3">Max</div>
+const QuotaScopes = ({quota}) => {
+  const scopes = _.get(quota, ['spec', 'scopes']);
+  if (_.isEmpty(scopes)) {
+    return <dd>-</dd>;
+  }
+  return scopes.map(scope => {
+    const scopeObj = _.get(quotaScopes, scope);
+    return scopeObj ?
+      <dd key={scope}>
+        <div className="co-resource-quota-scope__label">{scopeObj.label}</div>
+        <div className="co-resource-quota-scope__description">{scopeObj.description}</div>
+      </dd>
+      : <dd key={scope} className="co-resource-quota-scope__label">{scope}</dd>;
+  });
+};
+
+const showCharts = resourceTypes => {
+  const chartResourceTypes = ['requests.cpu', 'cpu', 'limits.cpu', 'requests.memory', 'memory', 'limits.memory'];
+  return _.intersection(resourceTypes, chartResourceTypes).length > 0;
+};
+
+const Details = ({obj: rq}) => {
+  const resourceTypes = getResourceTypes(rq);
```

Comment:
Hey, Dave. This isn't a huge issue because the list should never be too hard, but I still think we should avoid iterating over the array again and again to check if the resource type is there. It shouldn't be necessary. There are a few pretty easy ways to fix this.

1. Check `spec.hard[key]` to check if key is present.
2. Have `getResourceTypes()` return an `OrderedSet`.
3. Create a `Set` here and replace the `_.includes` checks.

For instance,

```suggestion
  const resourceTypes = getResourceTypes(rq);
  const resourceTypesSet = new Set(resourceTypes);
```

Then...

```js
  if (resourceTypesSet.has('requests.cpu')) {
```

etc

This prevents iterating over the entire array every time to check if the type exists.

Reply:
Ok, fixed, using a Set